addons/point_of_sale/models/pos_order.py

```diff
@@ -446,6 +448,14 @@ def create(self, vals_list):
         for vals in vals_list:
             session = self.env['pos.session'].browse(vals['session_id'])
             vals = self._complete_values_from_session(session, vals)
+            if vals.get('mobile'):
+                related_country = self.env['res.country']
+                if vals.get('partner_id'):
+                    related_country = self.env['res.partner'].browse(vals['partner_id']).country_id
+                else:
+                    related_country = self.env.company.country_id
+                vals['mobile'] = self._phone_format(number=vals.get('mobile'),
+                        country=related_country)
```

Comment:
```suggestion
            related_country = False
            if vals.get('mobile'):
                related_country = self.env['res.partner'].browse(vals['partner_id']).country_id if vals.get('partner_id') else self.env.company.country_id
                vals['mobile'] = self._phone_format(number=vals.get('mobile'), country=related_country)
```

I would do something like this